src/haz3lweb/view/LangDoc.re

```diff
@@ -1902,6 +1915,50 @@ let get_doc =
             LangDocMessages.funapp_exp_coloring_ids,
           );
         };
+      | DeferredAp(x, args) =>
+        let x_id = List.nth(x.ids, 0);
+        let supplied_id = (-1);
+        let deferred_id = (-2);
+        let (doc, options) =
+          LangDocMessages.get_form_and_options(
+            LangDocMessages.deferred_funapp_exp_group,
+            docs,
+          );
+        switch (mode) {
+        | MessageContent(_) =>
+          get_message(
+            doc,
+            options,
+            LangDocMessages.deferred_funapp_exp_group,
+            Printf.sprintf(
+              Scanf.format_from_string(doc.explanation.message, "%i%i%i"),
+              x_id,
+              supplied_id,
+              deferred_id,
+            ),
+            LangDocMessages.deferred_funapp_exp_coloring_ids(
+              ~x_id,
+              ~supplied_id,
+              ~deferred_id,
+            ),
+          )
+        | Colorings =>
```

Comment:
![image](https://github.com/hazelgrove/hazel/assets/108375845/7bff223c-0067-4972-ac11-320e2a3aa720)
Color highlighting is done though the implementation in `LangDoc.re` has become really hacky... It seems that each term should have a different color and this implementation circumvents `ColorSteps.get_color`